apps/api/src/app/tenant/usecases/create-tenant/create-tenant.usecase.ts

```diff
@@ -0,0 +1,28 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { TenantRepository } from '@novu/dal';
+import { CreateTenantCommand } from './create-tenant.command';
+
+@Injectable()
+export class CreateTenant {
+  constructor(private tenantRepository: TenantRepository) {}
+
+  async execute(command: CreateTenantCommand) {
+    const tenantExist = await this.tenantRepository.findOne({
+      _environmentId: command.environmentId,
+      identifier: command.identifier,
+    });
+
+    if (tenantExist) {
+      throw new BadRequestException(
```

Comment:
As far as I know, we do not want to make duplication between the environments.

Reply:
Maybe throw `ConflictException`?